lib/stormpath-sdk/client.rb

```diff
@@ -20,18 +20,18 @@ class Client
     include Stormpath::Util::Assert
     include Stormpath::Resource::Associations
 
-    attr_reader :data_store, :application
+    attr_reader :data_store, :application, :api_key
```

Comment:
I agree that we shouldn't have it on the client. Can you check my last two commits on this PR. 

I removed the api_key from the HttpClientRequestExecutor and api key is now passed as parameter when the data_store is initialized in the client. Also when the execute_request is called from the data_store api_key is passed as a parameter.

This way the api_key will be only available via data_store, and RequestExecutor doesen't need the api_key as instance variable. Also the test suite needed minor modifications and the tests where passing localy.

what do you think on this approach?